### PR TITLE
Release for Cosmos spark connector 4.5.0

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+#### Key Bug Fixes
+* Suppressing query plan retrieval in Spark queries (where they never would be necessary) to suppress I/O calls to teh gateway and improve latency for query execution. - See [PR 25668](https://github.com/Azure/azure-sdk-for-java/pull/25668)
+* Fixed a bug resulting in not being able to use views defined in the CosmosCatalog after cluster restart. Error observed in this case was `com.databricks.backend.common.rpc.DatabricksExceptions$SQLExecutionException: org.json4s.package$MappingException: unknown error`. - See [PR 25908](https://github.com/Azure/azure-sdk-for-java/pull/25908)
+* Modified bulk execution settings (maxPendingActions, I/O Thread count factor and 408-retry backoff) to improve default experience in scenarios with significantly more physical partition sin Cosmos than Spark Executor Cores. - See [PR 25914](https://github.com/Azure/azure-sdk-for-java/pull/25914)
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md:9]: typo: "more physical partitions in"

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/Samples/Python/NYC-Taxi-Data/01_Batch.ipynb [hunk 2]
```diff
@@ -323,7 +323,7 @@
     "  \"spark.cosmos.accountKey\": cosmosMasterKey,\n",
     "  \"spark.cosmos.database\": \"SampleDatabase\",\n",
     "  \"spark.cosmos.container\": \"GreenTaxiRecords\",\n",
-    "  \"spark.cosmos.read.partitioning.strategy\": \"Default\",\n",
+    "  \"spark.cosmos.read.partitioning.strategy\": \"Restrictive\",#IMPORTANT - any other partitioning strategy will result in indexing not being use to count - so latency and RU would spike up\n",
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/Samples/Python/NYC-Taxi-Data/01_Batch.ipynb:326]: will the situation change in future? 
should we always recommend customer to use "restrictive" now?

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/configuration-reference.md [hunk 3]
```diff
@@ -71,6 +71,7 @@ Used to influence the json serialization/deserialization behavior
 | :---        |    :----   |         :--- | 
 | `spark.cosmos.read.partitioning.strategy`     | `Default`    | The partitioning strategy used (Default, Custom, Restrictive or Aggressive) |
 | `spark.cosmos.partitioning.targetedCount`      | None    | The targeted Partition Count. This parameter is optional and ignored unless strategy==Custom is used. In this case the Spark Connector won't dynamically calculate number of partitions but stick with this value.  |
+| `spark.cosmos.partitioning.feedRangeFilter`      | None    | Can be used to scope the query to a single logical Cosmos partition (or a subset of logical partitions). This parameter is optionalIf provided the partitioning strategy will be modified - only partitions for the scoped logical partitions will be created. So the main benefit of this config option is to reduce the necessary SparkTasks/Partitions. |
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/configuration-reference.md:74]: nit: optionally -> optional if

**Review by xinlian12 — APPROVED** [review]
LGTM. Just some minor comments

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/Samples/Python/NYC-Taxi-Data/01_Batch.ipynb (hunk 2 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/Samples/Python/NYC-Taxi-Data/01_Batch.ipynb:326]: Just when using custom queries with aggregates like COUNT(0). Query team won't be able to use index - instead this becomes a full table scan if the EPK range doesn't align with a physical partition.
Need to get details from Query team - but our sample should for the "efficient count" sample definitely avoid the table scan

**Pull request merged**